jaxrs/src/main/java/feign/jaxrs/JAXRSContract.java

```diff
@@ -124,7 +126,12 @@ protected boolean processAnnotationsOnParameter(MethodMetadata data, Annotation[
     boolean isHttpParam = false;
     for (Annotation parameterAnnotation : annotations) {
       Class<? extends Annotation> annotationType = parameterAnnotation.annotationType();
-      if (annotationType == PathParam.class) {
+      // masc20180327. parameter with unsupported jax-rs annotations should not be passed as body params.
+      // this will prevent interfaces from becoming unusable entirely due to single (unsupported) endpoints.
+      // https://github.com/OpenFeign/feign/issues/669
+      if (annotationType == Suspended.class || annotationType == Context.class) {
```

Comment:
Can we use a String check here?

`annotationType.getClass().equals("javax.ws.rs.container.Suspended")`
Why? To prevent class def not found for anyone that still wanna use jax-rs 1

Reply:
yes that may work, but it's ugly and prone to breakage on maintenance so I won't supply this. I'd rather have `feign-jaxrs-2`. also I don't see any reason (for anyone) to even include `jax-rs-1`.

Reply:
how do you plan to support `jax-rs-2` in feign in general?

Reply:
> it's ugly and prone to breakage on maintenance
agree

may be a `try-catch` for the `ClassDefNotFound`

Reply:
@velo is it universally predictable when jvm's throw this?

Reply:
I would assume this is thrown by jvm before `processAnnotationsOnParameter` is invoked even, but I could be wrong.

Reply:
I made a PR with an "example" and the exception as well
https://github.com/masc3d/feign/pull/1

Reply:
Not sure if it's overkill or not, but you could always use the maven-invoker-plugin to test compatibility.